### ISSUE TITLE
fontforge/splinestroke.c:969: SplineStrokeAppendFixup: Assertion `mg < 1' failed.

### ISSUE THREAD

**Issue by ctrlcctrlv** [fontforge/fontforge#4053]
Arch Linux, 3890b4762567b84cd57b6d03fb382ad7fdb3ad50

cc: @skef

# Settings

* Circular, 45-45-0
* Setting &laquo;Remove Overlap&raquo; to &laquo;None&raquo; does not fix the problem.

# File

[J.sfd](https://github.com/fontforge/fontforge/files/3948568/J.zip)

# Complete output
```
[fred@pc Typewriter AA 2020]$ fontforge /tmp/J.sfd
Copyright (c) 2000-2019. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20190801
 Based on sources from 2019-11-26 12:18 UTC-ML-D-GDK3.
 Based on source from git with hash: 5c2dbec01dadefb7eef1537c65b40a64e8a8da31
PythonUI_Init()
copyUIMethodsToBaseTable()
skipping unqualified hotkey for widget g:0x55d54f0f2b70
skipping unqualified hotkey for widget g:0x55d54f0f2b70
skipping unqualified hotkey for widget g:0x55d54f0f2b70
skipping unqualified hotkey for widget g:0x55d54f0f2b70
skipping unqualified hotkey for widget g:0x55d54f0f2b70
skipping unqualified hotkey for widget g:0x55d54f0f2b70
fontforge: ../fontforge/splinestroke.c:969: SplineStrokeAppendFixup: Assertion `mg < 1' failed.
Aborted (core dumped)
```

**Comment by ctrlcctrlv** [comment]
The new stroker seems to not like the selected point in the below screenshot:

![2019-12-11-131749_2560x2101_scrot](https://user-images.githubusercontent.com/838783/70593577-871f7500-1c18-11ea-99e8-1a6ecd574c2f.png)

Moving it down even an imperceptible amount averts this crash.

**Comment by skef** [comment]
Probably some bug in `DoubleBackJC()`. These arbitrary open contours fill me with dread.

**Comment by skef** [comment]
I'm having trouble reproducing this one. Could you verify with the file? Note that it opens for me with Cubic Bezier rather than Spiro enabled.

**Comment by ctrlcctrlv** [comment]
This one also reproduces for me. It also opens for me in B&eacute;zier mode but that doesn't matter on my system.

**Comment by skef** [comment]
Ah, OK, not sure what I changed but I can reproduce now.

**Comment by skef** [comment]
Well, this isn't super-satisfying, but my fix for #4052 seems to fix this one as well.  Offhand I'm not sure why that is. The last spline (at the end of the j loop) is the closest to almost-but-not-quite-linear. But it doesn't seem like moving the first point would affect the last one so much. 

Try it out, though.

**Comment by ctrlcctrlv** [comment]
Confirmed, fixed by #4052
